app/lib/search/text_utils.dart

```diff
@@ -33,3 +34,12 @@ Iterable<String> splitForIndexing(String text) {
   if (text == null || text.isEmpty) return new Iterable.empty();
   return text.split(_separators).where((s) => s.isNotEmpty);
 }
+
+List<String> extractExactPhrases(String text) => _exactTermRegExp
+    .allMatches(text)
+    .map((m) => m.group(1).trim())
+    .where((s) => s.isNotEmpty)
+    .toList();
+
+bool matchesPhrase(String text, String phrase) =>
+    normalizeBeforeIndexing(text).contains(normalizeBeforeIndexing(phrase));
```

Comment:
This is an `O(3 * n)` for normalization plus another `O(n)` for `contains`. For larger README files this is really not good!

Reply:
I'll add a pre-processing step to reduce the time of this. For this check, we could concatenate the title, description and readme (strictly speaking there is a chance for false-positives, but that is probably not an issue in practice), and do the normalization at indexing time.